### PR TITLE
Add Boot Option details to boot order data source

### PR DESCRIPTION
**Pull request by kanth-d**
##### SUMMARY

Relates OR Closes #0000

##### ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request
- New Module Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

#### Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note

```

#### Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

#### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

### CODE REVIEW

**Comment by terraform-providers-svc** [comment]
Can one of the admins verify this patch?

File: redfish/provider/data_source_redfish_bios.go [hunk 1]
```diff
@@ -129,6 +138,12 @@ func (g *BiosDatasource) readDatasourceRedfishBios(d models.BiosDatasource) (mod
 		return d, diags
 	}
 
+	bootOptions, err := systems[0].BootOptions()
+	if err != nil {
+		diags.AddError("Error fetching boot", err.Error())
```
Comment (rounak-adhikary) [inline redfish/provider/data_source_redfish_bios.go:143]:
```suggestion
		diags.AddError("Error fetching boot options", err.Error())
```

File: redfish/provider/data_source_redfish_bios.go [hunk 2]
```diff
@@ -147,5 +162,59 @@ func (g *BiosDatasource) readDatasourceRedfishBios(d models.BiosDatasource) (mod
 	d.ID = types.StringValue(bios.ID)
 	d.Attributes, diags = types.MapValue(types.StringType, attributes)
 
+	bootOptionsList := make([]models.BiosBootOptions, 0)
+	for i := range bootOptions {
+		bootOptionsList = append(bootOptionsList, newBootOption(bootOptions[i]))
+	}
```
Comment (rounak-adhikary) [inline redfish/provider/data_source_redfish_bios.go:168]:
```suggestion
	for _, bootOption := range bootOptions {
		bootOptionsList = append(bootOptionsList, newBootOption(bootOption))
	}
```

File: redfish/provider/data_source_redfish_bios.go [hunk 3]
```diff
@@ -147,5 +162,59 @@ func (g *BiosDatasource) readDatasourceRedfishBios(d models.BiosDatasource) (mod
...
+		UefiDevicePath:      types.StringValue(input.UefiDevicePath),
+	}
+}
+
+// BootOptionsSchema is a function that returns the schema for Boot Options
+func BootOptionsSchema() map[string]schema.Attribute {
+	return map[string]schema.Attribute{
+		"boot_option_enabled": schema.BoolAttribute{
+			MarkdownDescription: "Enable or disable the boot device.",
+			Description:         "Enable or disable the boot device.",
```
Comment (rounak-adhikary) [inline redfish/provider/data_source_redfish_bios.go:191]:
```suggestion
			Description:         "Whether the boot device is enabled or not.",
```
Same for the MarkdownDescription

**Review by Krishnan-Priyanshu — APPROVED** [review]

**Review by Krishnan-Priyanshu — CHANGES_REQUESTED** [review]
please add coverage report and test for boot options

**Comment by kanth-d** [comment]
image

**Review by kanth-d — COMMENTED** [review]
Added test and coverage report screenshot

**Review by Krishnan-Priyanshu — APPROVED** [review]

**Review by rounak-adhikary — APPROVED** [review]
LGTM

**Review by Sakshi-dell — APPROVED** [review]

**Pull request merged**